### ISSUE TITLE
[Bug] "Incompatible World"

### ISSUE THREAD

**Issue by TheButterbrotMan** [quiqueck/BCLib#24]
### What happened?

Everytime when i try to load a world "detected an incompatible world". This happens without anything being changed.

### BCLib

bclib-1.4.7

### Fabric API

fabric-api-0.57.0+1.18.2

### Fabric Loader

0.14.8

### Minecraft

1.18.2

### Other Mods

https://www.curseforge.com/minecraft/modpacks/deathdusk (2.1.0)

**Comment by quiqueck** [comment]
What other Mods do you have installed? Maybe BYG?

This thread might help to configure your settup: quiqueck/BetterNether#34. Let me know if you need any assistance with this.

**Comment by TheButterbrotMan** [comment]
![Screenshot_214](https://user-images.githubusercontent.com/64548098/179866241-e29eff53-6a92-4029-a91d-49795de8fb11.png)

**Comment by TheButterbrotMan** [comment]
Yes, i have a lot of other mods. BYG is one if them.

**Comment by TheButterbrotMan** [comment]
> What other Mods do you have installed? Maybe BYG?
> 
> This thread might help to configure your settup: [quiqueck/BetterNether#34](https://github.com/quiqueck/BetterNether/issues/34). Let me know if you need any assistance with this.

Are you sure this is connected to my problem? I dont read there anything about a everytime coming back fix screen. But i can try the fix.

**Comment by quiqueck** [comment]
My second/third reply in that issue (https://github.com/quiqueck/BetterNether/issues/34#issuecomment-1178356709) explains the issue you are seeing.

**Comment by TheButterbrotMan** [comment]
> What other Mods do you have installed? Maybe BYG?
> 
> This thread might help to configure your settup: [quiqueck/BetterNether#34](https://github.com/quiqueck/BetterNether/issues/34). Let me know if you need any assistance with this.

Maybe i need assistance. So i add nether-biomes.json5 to my configs. Then i disable byg biomes? and then add them manually into bclib?

Im not english native, its a little bit hard to follow your conversation xD

**Comment by quiqueck** [comment]
No worries :)

You need to look for all the files in your Minecraft Folder (on windows it is located in the folder `%appdata%\.minecraft\`). They should all exist, as the Mods (BCLib and BYG) generate them on the first start with default settings.

### 1. Disable the BYG BiomeSources 
This will let BCLib take control, and load the correct Biomes.

* For the Nether open **config/byg/nether-biomes.json5** and change `"forceBYGNetherBiomeSource": true,` to `"forceBYGNetherBiomeSource": false,`

* For the End open **config/byg/end-biomes.json5** and change `"forceBYGEndBiomeSource": true,` to `"forceBYGEndBiomeSource": false,`

### 2. Tell BCLib to include all Nether Biomes from BYG. 
Open **config/bclib/biomes.json**. You should see a block of text that looks like this:
```json
{
  "force_include": {
    "end_land_biomes": [],
    "end_void_biomes": [],
    "nether_biomes": []
  },
```
 You will need to edit the `nether_biomes`-line. For that, find the file that is called **config/byg/nether-biomes.json5**. In that file, you will find entries like 
```json
{
    "data": "byg:weeping_mire",
    "weight": 2
}
```
Copy all the `byg:...` entries to the `nether_biomes`-line. For my (on my dev box, there currently is only 1.19) it had the following Biomes:

```json
{
  "force_include": {
    "end_land_biomes": [],
    "end_void_biomes": [],
    "nether_biomes": ["byg:weeping_mire", "byg:embur_bog", "byg:brimstone_caverns", "byg:subzero_hypogeal", "byg:warped_desert", "byg:magma_wastes", "byg:quartz_desert", "byg:crimson_gardens", "byg:glowstone_gardens", "byg:sythian_torrids", "byg:wailing_garth", "byg:arisian_undergrowth"]
  },
```
That might work for you as well. Not sure if BYG did add/remove Biomes from 1.18.2 to 1.19

### 3. Tell BCLib to include all End Biomes from BYG. 
That works the same as the Nether Biomes. Except you will get the End-Biomes from **config/byg/end-biomes.json5**, and you'll have to add them to the `end_land_biomes`or `end_void_biomes` line. For me this is how the result looks:
```json
{
  "force_include": {
    "end_land_biomes": ["byg:ivis_fields", "byg:nightshade_forest", "byg:bulbis_gardens", "byg:shulkren_forest", "byg:cryptic_wastes", "byg:imparius_grove"],
    "end_void_biomes": ["byg:ethereal_islands", "byg:viscal_isles"],
    "nether_biomes": ["byg:weeping_mire", "byg:embur_bog", "byg:brimstone_caverns", "byg:subzero_hypogeal", "byg:warped_desert", "byg:magma_wastes", "byg:quartz_desert", "byg:crimson_gardens", "byg:glowstone_gardens", "byg:sythian_torrids", "byg:wailing_garth", "byg:arisian_undergrowth"]
  },
```

Hope this helps a bit :)

**Comment by TheButterbrotMan** [comment]
That helps a lot. Thank you!

**Comment by TheButterbrotMan** [comment]
Is there any wiki about the config from bclib? I saw it has a lot of config options. Maybe some of them useful for modpackcreators.

**Comment by quiqueck** [comment]
Not yet, a lot is currently changing concerning the configs (many options will move to DataPacks for example)

**Comment by Abalieno** [comment]
I've gone through this (disabling biomesource in the two byg files, and moving all biomes to bclib biomes.json) but nothing was fixed.

I still get the constant Guardian screen.

![ykyid](https://user-images.githubusercontent.com/4656436/188348197-a5fe9add-a46d-4d53-8340-993371e7f2fd.jpg)

I guess the problem might be that there's again some other mod overwriting it, in this case Incendium, Nullscape and Yttr?

Is there no way to have these mods together? No other fix than removing them?

**Comment by quiqueck** [comment]
You might also need to disable the checks as discussed here:
https://github.com/quiqueck/BetterNether/issues/34#issuecomment-1178699409

**Comment by Abalieno** [comment]
Well, yes.

Setting "repairBiomesOnLoad" to false removes the screen. I'm just wondering if things actually work or not.

Ive verified Yttr doesn't trigger the screen, but Incendium and Nullscape do. I'm just wondering if there's a way to have all these mods show their own content.

From what I read, byg was overriding biome generation, and Better Nether wouldn't work. So turning byg off and offload byg own biomes to BCLib would properly generate both the "Better" biomes along with byg ones.

But now I wonder if Incendium could also take control in its own way. I guess I'll have to test and see at some point.

**Comment by quiqueck** [comment]
Ah, ok.  
Yes Incendium will also overwrite the biome source and disable the better* Biomes. I am not sure if BYG was able to do proper Datapack handling in 1.18.2. But (with a Datapack Mod Like Incendium) it may work if you let byg take over and manually add the better* biomes to their Biome Sources.

**Comment by Abalieno** [comment]
I suppose it doesn't change that Incendium is used in its mod format, rather than datapack, as the content would still be similar?

In any case, wasn't the issue that if byg took over BCLib, then Better Nether/End wouldn't generate?

As seen above: https://github.com/quiqueck/BetterNether/issues/45

...If using Incendium doesn't generate Incendium biomes then I guess Incendium isn't doing the thing.
